### PR TITLE
Force package names to contain an organization prefix

### PR DESCRIPTION
**Pull request by stevededalus**
Add support for one level of nesting in package registries.

### CODE REVIEW

File: api/python/t4/packages.py [hunk 1]
```diff
@@ -116,6 +117,13 @@ def _clone(self):
 class Package(object):
     """ In-memory representation of a package """
 
+    @staticmethod
+    def validate_package_name(name):
+        """ Verify that a package name is two alphanumerics strings separated by a slash."""
+        if not re.match(r"[a-zA-Z0-9_\\-]+/[a-zA-Z0-9_\\-]+$", name):
```
Comment (dimaryaz) [inline api/python/t4/packages.py]: Either `r` prefix or double backslash; this one allows backslashes in the package name.
Also, make the regex a global const in `util.py` or something? Could pre-compile it, too.

File: api/python/t4/api.py [hunk 2]
```diff
@@ -137,7 +139,14 @@ def list_packages(registry=None):
 
     registry_url = urlparse(registry)
     if registry_url.scheme == 'file':
-        return os.listdir(parse_file_url(registry_url))
+        # There's probably a cleaner glob.glob(parse_file_url(registry_url)+'/*')
```
Comment (dimaryaz) [inline api/python/t4/api.py]: Here's one way:
```
registry_dir = pathlib.Path(parse_file_url(registry_url))
return [str(x.relative_to(registry_dir)) for x in registry_dir.glob('*/*')]
```

File: api/python/t4/packages.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (stevededalus) [reply api/python/t4/packages.py]: oh, the `-` is just to support packages with `-` in the name i.e. `quilt-alpha/foo`. I originally had `\w` but noticed a lot of our test packages have dashes:
https://alpha.quiltdata.com/browse/.quilt/named_packages/

[the remark this answers is not in the document]
Comment (dimaryaz) [reply api/python/t4/packages.py]: You can use `\w` inside of `[...]`, btw

**Pull request merged**